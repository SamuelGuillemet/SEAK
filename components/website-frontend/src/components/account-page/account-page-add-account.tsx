import { useState } from 'react';
import toast from 'react-hot-toast';

import { useQueryClient } from '@tanstack/react-query';

import Register from '@/components/register';
import { Button } from '@/components/ui/button';
import { Dialog, DialogContent, DialogTrigger } from '@/components/ui/dialog';

export function AccountPageAddAccount(): React.JSX.Element {
  const query = useQueryClient();
  const [isOpen, setIsOpen] = useState(false);

  const onSuccess = () => {
    toast.success('Compte ajouté avec succès !');
    setIsOpen(false);
    query.invalidateQueries({ stale: true });
  };

  return (
    <Dialog
      open={isOpen}
      onOpenChange={setIsOpen}
    >
      <DialogTrigger asChild>
        <Button className='place-self-end m-4'>Ajouter un compte</Button>
      </DialogTrigger>
      <DialogContent>
        <Register onSuccess={onSuccess} />
      </DialogContent>
    </Dialog>
  );
}
