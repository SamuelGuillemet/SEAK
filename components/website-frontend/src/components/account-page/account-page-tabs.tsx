import Link from 'next/link';
import { useRouter } from 'next/router';
import type { Session } from 'next-auth';
import { useSession } from 'next-auth/react';

import { SecurityScopes } from '@/openapi-codegen/apiSchemas';
import { pages } from '@/utils/pages';

type LinksType = Array<{
  label: string;
  href: string;
  scopes: SecurityScopes[];
}>;

export const links: LinksType = [
  { label: 'Accueil', href: pages.account.index, scopes: ['user', 'admin'] },
  { label: 'Comptes utilisateurs', href: pages.account.users, scopes: ['admin'] }
];

export function findActiveLink(pathname: string) {
  return links.find((link) => link.href.search(pathname) !== -1);
}

export function filterLinks(session: Session | null) {
  if (session) {
    const userRoles = session.scopes;
    return links.filter((link) => link.scopes.some((role) => userRoles.includes(role)));
  } else {
    return [];
  }
}

export function Tabs() {
  const { pathname } = useRouter();
  const { data: session } = useSession();

  const activeLink = findActiveLink(pathname);
  const filteredLinks = filterLinks(session);

  return (
    <div className='flex justify-center pr-2 md:border-r md:border-b-0 border-b md:mr-4 mb-4 md:mb-0'>
      <div className='w-full max-w-md md:w-max flex justify-start items-center md:items-start md:flex-col flex-row mb-2 md:mb-0'>
        {filteredLinks.map((link) => (
          <Link
            href={link.href}
            key={link.href}
            className={`px-3 py-2 text-sm w-full font-medium ${activeLink === link ? 'bg-slate-400 dark:bg-gray-800' : 'text-gray-900 hover:bg-slate-300 dark:text-gray-300 dark:hover:bg-gray-700'}`}
          >
            {link.label}
          </Link>
        ))}
      </div>
    </div>
  );
}
