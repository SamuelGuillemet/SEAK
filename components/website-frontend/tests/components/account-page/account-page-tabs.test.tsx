import { render } from "@testing-library/react";

import { Tabs } from "@/components/account-page/account-page-tabs";

describe("Tabs", () => {
  it("should render the tabs", () => {
    const { container } = render(<Tabs />);
    expect(container).toMatchSnapshot();
  });
});
