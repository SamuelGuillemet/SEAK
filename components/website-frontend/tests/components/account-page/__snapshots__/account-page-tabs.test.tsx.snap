// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tabs > should render the tabs 1`] = `
<div>
  <div
    class="flex justify-start pr-2 md:border-r md:border-b-0 border-b md:mr-4 mb-4 md:mb-0"
  >
    <div
      class="w-full max-w-md md:w-max flex justify-start items-center md:items-start md:flex-col flex-row mb-2 md:mb-0 gap-4"
    >
      <a
        href="/account"
      >
        <div
          class="flex items-center gap-2 p-2 rounded-md hover:bg-gray-200 dark:hover:bg-gray-700 "
        >
          <svg
            class="h-6 w-6"
            fill="none"
            height="15"
            viewBox="0 0 15 15"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M7.5 0.875C5.49797 0.875 3.875 2.49797 3.875 4.5C3.875 6.15288 4.98124 7.54738 6.49373 7.98351C5.2997 8.12901 4.27557 8.55134 3.50407 9.31167C2.52216 10.2794 2.02502 11.72 2.02502 13.5999C2.02502 13.8623 2.23769 14.0749 2.50002 14.0749C2.76236 14.0749 2.97502 13.8623 2.97502 13.5999C2.97502 11.8799 3.42786 10.7206 4.17091 9.9883C4.91536 9.25463 6.02674 8.87499 7.49995 8.87499C8.97317 8.87499 10.0846 9.25463 10.8291 9.98831C11.5721 10.7206 12.025 11.8799 12.025 13.5999C12.025 13.8623 12.2376 14.0749 12.5 14.0749C12.7623 14.075 12.975 13.8623 12.975 13.6C12.975 11.72 12.4778 10.2794 11.4959 9.31166C10.7244 8.55135 9.70025 8.12903 8.50625 7.98352C10.0187 7.5474 11.125 6.15289 11.125 4.5C11.125 2.49797 9.50203 0.875 7.5 0.875ZM4.825 4.5C4.825 3.02264 6.02264 1.825 7.5 1.825C8.97736 1.825 10.175 3.02264 10.175 4.5C10.175 5.97736 8.97736 7.175 7.5 7.175C6.02264 7.175 4.825 5.97736 4.825 4.5Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
          <span
            class="md:text-lg font-semibold"
          >
            Mon compte
          </span>
        </div>
      </a>
      <a
        href="/account/users"
      >
        <div
          class="flex items-center gap-2 p-2 rounded-md hover:bg-gray-200 dark:hover:bg-gray-700 "
        >
          <svg
            class="h-6 w-6"
            fill="none"
            height="15"
            viewBox="0 0 15 15"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M1.5 5.25C1.91421 5.25 2.25 4.91421 2.25 4.5C2.25 4.08579 1.91421 3.75 1.5 3.75C1.08579 3.75 0.75 4.08579 0.75 4.5C0.75 4.91421 1.08579 5.25 1.5 5.25ZM4 4.5C4 4.22386 4.22386 4 4.5 4H13.5C13.7761 4 14 4.22386 14 4.5C14 4.77614 13.7761 5 13.5 5H4.5C4.22386 5 4 4.77614 4 4.5ZM4.5 7C4.22386 7 4 7.22386 4 7.5C4 7.77614 4.22386 8 4.5 8H13.5C13.7761 8 14 7.77614 14 7.5C14 7.22386 13.7761 7 13.5 7H4.5ZM4.5 10C4.22386 10 4 10.2239 4 10.5C4 10.7761 4.22386 11 4.5 11H13.5C13.7761 11 14 10.7761 14 10.5C14 10.2239 13.7761 10 13.5 10H4.5ZM2.25 7.5C2.25 7.91421 1.91421 8.25 1.5 8.25C1.08579 8.25 0.75 7.91421 0.75 7.5C0.75 7.08579 1.08579 6.75 1.5 6.75C1.91421 6.75 2.25 7.08579 2.25 7.5ZM1.5 11.25C1.91421 11.25 2.25 10.9142 2.25 10.5C2.25 10.0858 1.91421 9.75 1.5 9.75C1.08579 9.75 0.75 10.0858 0.75 10.5C0.75 10.9142 1.08579 11.25 1.5 11.25Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
          <span
            class="md:text-lg font-semibold"
          >
            Comptes utilisateurs
          </span>
        </div>
      </a>
    </div>
  </div>
</div>
`;
